src/base/http/server.cpp

```diff
@@ -112,26 +123,20 @@ void Server::dropTimedOutConnection()
     }
 }
 
-#ifndef QT_NO_OPENSSL
 bool Server::setupHttps(const QByteArray &certificates, const QByteArray &key)
 {
-    QSslKey sslKey(key, QSsl::Rsa);
-    if (sslKey.isNull())
-        sslKey = QSslKey(key, QSsl::Ec);
-
-    const QList<QSslCertificate> certs = QSslCertificate::fromData(certificates);
-    const bool areCertsValid = !certs.empty() && std::all_of(certs.begin(), certs.end(), [](const QSslCertificate &c) { return !c.isNull(); });
-
-    if (!sslKey.isNull() && areCertsValid) {
-        m_key = sslKey;
-        m_certificates = certs;
-        m_https = true;
-        return true;
-    }
-    else {
+    if (!isCertValid(certificates) || !isKeyValid(key)) {
```

Comment:
Using these helpers you parse certificates/keys data twice in this method. I would refactor it.

Reply:
>I would refactor it.

I think using boost::optional would be a perfect fit here.

Reply:
Don't understand how you want use it...

Reply:
PR updated, this way the cert/key loading code can keep in just one function.